src/store-api/src/metadata.rs

```diff
@@ -639,42 +639,34 @@ impl RegionMetadataBuilder {
             if column_meta.column_schema.name == column_name {
                 ensure!(
                     column_meta.column_schema.data_type.is_string(),
-                    SetFulltextOptionsSnafu {
-                        column_name: column_name.to_string(),
-                        reason: format!(
-                            "column {} is not a string type, but {:?}",
-                            column_name, column_meta.column_schema.data_type
-                        ),
+                    InvalidColumnOptionSnafu {
+                        column_name,
+                        msg: format!(
+                            "expect a string type, but got {}",
+                            column_meta.column_schema.data_type
+                        )
```

Comment:
nit: let's keep the same style
```suggestion
                        msg: "FULLTEXT index only supports string type"
```